src/legacy/server/index_patterns/service/lib/resolve_time_pattern.ts

```diff
@@ -34,43 +42,40 @@ import { callIndexAliasApi } from './es_api';
  *                            and the indices that actually match the time
  *                            pattern (matches);
  */
-export async function resolveTimePattern(callCluster, timePattern) {
+export async function resolveTimePattern(callCluster: APICaller, timePattern: string) {
   const aliases = await callIndexAliasApi(callCluster, timePatternToWildcard(timePattern));
 
-  const allIndexDetails = chain(aliases)
-    .reduce((acc, index, indexName) => acc.concat(
-      indexName,
-      Object.keys(index.aliases || {})
-    ), [])
+  const allIndexDetails: IndexDetail[] = chain(aliases)
+    .reduce(
+      (acc, index, indexName) => acc.concat(indexName, Object.keys(index.aliases || {})),
+      [] as string[]
+    )
```

Comment:
The `.reduce(` call here is throwing a number of errors related to lodash that I don't understand- maybe we need to split it out into `chain(aliases.reduce(...))`?